### PR TITLE
[native_assets_cli] `BuildOutput` dependencies per asset and asset type

### PR DESCRIPTION
**Pull request by dcharkes**
Bug:

* https://github.com/dart-lang/native/issues/1208

### CODE REVIEW

File: pkgs/native_assets_builder/test_data/add_asset_link/hook/link.dart [hunk 1]
```diff
@@ -7,17 +7,15 @@ import 'package:native_assets_cli/native_assets_cli.dart';
...
-        NativeCodeAsset(
-          package: 'add_asset_link',
-          name: 'dylib_add_link',
-          linkMode: builtDylib.linkMode,
-          os: builtDylib.os,
-          architecture: builtDylib.architecture,
-          file: builtDylib.file,
-        ),
-      )
-      ..addDependency(config.packageRoot.resolve('hook/link.dart'));
```
Comment (mosuem) [inline pkgs/native_assets_builder/test_data/add_asset_link/hook/link.dart:21]: Why remove dependencies in link hooks?

Comment (dcharkes) [reply pkgs/native_assets_builder/test_data/add_asset_link/hook/link.dart:21]: The Dart files don't need to be in the dependencies anymore.

Comment (mosuem) [reply pkgs/native_assets_builder/test_data/add_asset_link/hook/link.dart:21]: Forgot about that PR..

**Comment by dcharkes** [comment]
As discussed in https://github.com/dart-lang/native/issues/1207#issuecomment-2250343938, we don't want assets per dependency.

**Pull request closed without merging**